### PR TITLE
Fix some npc services

### PR DESCRIPTION
**Pull request by olanti-p**
#### 1. Fix an exploit with refugee center Broker
Repro steps:
1. Get them in debt by asking for items as mission reward, but don't take any
2. Bring them 1 jerky. They'll pay for jerky and repay their debt on top of that, but won't write the debt off.
3. Repeat 2 until you're drowning in money

#### 2. Remove usages of `u_spend_cash` dialogue effect
At some point all it did was simply draw money from pc's ATM account, but now it forwards to interactive barter screen. As such, any service that uses it can be received "for free" by pressing `Esc` when prompted to pay with items, since effects can't prevent  other effects in the same "chain" from firing.

This exploit affects the following NPCs:
* Tacoma nurse (heals pc and their party)
* Tacoma lumberjack (sells logs)
* Survivor with PREPPER_2 background story (sells survivor map)
    
I decided to completely switch nurse & lumberjack to accept Merch only. That means, player will have to handle huge amounts of currency, but that won't be as painful after #830.

Survivor is a bit tricky: it may be a potential day-one ally stuck in the same shelter as pc, so it wouldn't make sense for them to ask for any faction currency. It makes even less sense for them to rely on ATM account though, considering their story. I decided to replace the whole purchase option with an alternative "persuade" option. Survivor maps are pretty common, this shouldn't be _too_ OP.

#### 3. Remove Tacoma bartender's "What do you have on tap?" dialogue option
It's just bad. You can buy all the booze you would want via regular trade, and this separate option:
* Looks extremely confusing as it brings you to the regular trade screen anyway
* Only allows buying one item at a time
* Once you buy a bottle, you recognize the option for the elaborate scam it is:
image

I tried to fix it by switching price to Merch and using glasses as containers to make it at least passable as a pure RP option, but the drink spills out of glasses... It'll need additional code for bartender to spawn them on the counter (or a lumberjack-level horrible hardcoded hack), not sure if that's worth the effort.

#### 4. Use "[N Merch]" tag everywhere NPC asks for Merch as payment
Pretty straight change here.
The merc asks for "FMCN", the barber asks for dollars, while both actually consume an item named `Merch`.

### Describe alternatives you've considered
Ideally there would be a special response function (or a trial) that would prompt player to give up specified sum of items, or pay specified sum in faction's currency, but I feel existing crutchy solution should be enough for now. But it's the first time I'm working with dialogue, so maybe I'm missing something.
Also, with higher demand for Merch in game, would it make sense to bump amount of faction currency the traders offer?

[WIP][CR] Because of the above points, and in case someone will have a better idea on how to deal with those NPCs.

### CODE REVIEW

**Comment by olanti-p** [comment]
I've split off trade UI hint into its own PR (#873), as this one may take a bit to finish.

File: data/json/npcs/Backgrounds/prepper_2.json [hunk 1]
```diff
@@ -96,23 +105,45 @@
...
-    "dynamic_line": "Nice try.  You want the maps, you pay up.",
+    "dynamic_line": "Nice try.  But tell you what: if you're capable enough to stay alive through the week, I'll consider it.",
     "responses": [
       {
-        "text": "Fine.  What was it you were saying before?",
-        "effect": { "npc_add_effect": "player_BGSS_SAIDNO", "duration": 31536000 },
+        "text": "I'll hold you to this.",
+        "//": "Player can try again in a week",
+        "effect": [
+          { "npc_add_effect": "player_BGSS_SAIDNO", "duration": 604800 },
```
Comment (Coolthulhu) [inline data/json/npcs/Backgrounds/prepper_2.json:116]: Not strongly related to the PR, but those raw turn durations look terrible.

File: src/npctrade.cpp [hunk 2]
```diff
@@ -481,6 +481,15 @@ int trading_window::get_var_trade( const item &it, int total_count )
...
+                                     _( "Hint: You can buy up to %d with your current balance." ),
+                                     std::min( amount_hint, total_count ) ) );
+    } else if( amount_hint < 0 ) {
+        popup_input.description( string_format(
+                                     _( "Hint: You'll need to offer %d to even out the deal." ),
+                                     -amount_hint ) );
+    }
     popup_input.title( title ).edit( how_many );
     if( popup_input.canceled() || how_many <= 0 ) {
         return -1;
```
Comment (Coolthulhu) [inline src/npctrade.cpp:495]: Not caused here, but `cata::optional<int>` would be a minor improvement.

**Comment by olanti-p** [comment]
Ready.

I had some thoughts on making `u_has_cash`, `u_spend_cash` and `npc_service` work with faction currency (e.g. make them convert specified value to some amount of NPC's faction currency according to its value), but then realized we have very little dialog that could benefit from it, so I'd rather work on something more impactful instead.

**Pull request merged**